lib/migrate/Migrator.js

```diff
@@ -240,32 +240,23 @@ class Migrator {
     }
   }
 
-  _isLocked(trx) {
-    const tableName = getLockTableName(this.config.tableName);
-    return getTable(this.knex, tableName, this.config.schemaName)
-      .transacting(trx)
-      .forUpdate()
-      .select('*')
-      .then((data) => data[0].is_locked);
-  }
-
   _lockMigrations(trx) {
     const tableName = getLockTableName(this.config.tableName);
     return getTable(this.knex, tableName, this.config.schemaName)
       .transacting(trx)
-      .update({ is_locked: 1 });
+      .where('is_locked', '=', 0)
```

Comment:
This does not work. There can be two processes at the same time arriving to this row and both will return one row and both then updates the lock value to 1.

Reply:
forUpdate exactly causes DB to prevent that from happening.